ui/components/multichain/pages/permissions-page/connection-list-item.js

```diff
@@ -25,14 +26,21 @@ import {
 } from '../../../component-library';
 import { getURLHost } from '../../../../helpers/utils/util';
 import { SnapIcon } from '../../../app/snaps/snap-icon';
-import { getPermittedChainsForSelectedTab } from '../../../../selectors';
+import { getAllPermittedChainsForSelectedTab } from '../../../../selectors';
 
 export const ConnectionListItem = ({ connection, onClick }) => {
   const t = useI18nContext();
   const isSnap = connection.subjectType === SubjectType.Snap;
-  const connectedNetworks = useSelector((state) =>
-    getPermittedChainsForSelectedTab(state, connection.origin),
+  const _permittedChains = useSelector((state) =>
+    getAllPermittedChainsForSelectedTab(state, connection.origin),
   );
+  const permittedChains = _permittedChains.filter((caipChainId) => {
+    if (caipChainId === KnownCaipNamespace.Wallet) {
+      return false;
+    }
```

Comment:
maybe try catch is the better pattern here

Reply:
working on this

Reply:
[308911a](https://github.com/MetaMask/metamask-extension/pull/29887/commits/308911ab0990a99760609a4023da88fec0b9f6f6)